src/extension.js

```diff
@@ -70,9 +90,61 @@ const supportedLanguages = [
   'yaml',
 ];
 
+/**
+ * Provides a promise that resolves as soon as there is an active text editor
+ * with a document open on a language this extension supports.
+ *
+ * @returns {Promise<TextDocument>}
+ */
+function waitForActiveSupportedDocument() {
+  if (
+    !window.activeTextEditor
+    || !supportedLanguages.includes(window.activeTextEditor.document.languageId)
```

Comment:
I believe `supportedLanguages` should not be hard coded because of 3rd-party plugins. But it can be done in next minor version, I'd like to PR after v3 released later.